src/calculatePlayerMoves.ts

```diff
@@ -0,0 +1,269 @@
+// There has to be other player blocking the way in selected direction, wall behind it (to block linear jump)
+// and no walls to prevent jumping diagonally.
+const checkDiagonalFromOneSide = (
+  coordinatesSquare: Coordinates,
+  coordinatesPlayer: Coordinates,
+  board: Board,
+  startFromHorizontal: boolean
+): boolean => {
+  const square = getSquareByCoordinates(coordinatesSquare, board);
+  if (square.type !== SquareType.Player) throw new Error("Square is not of type 'Player'");
+  // When targeted square is occupied by player
+  if (square.playerId) return false;
+
+  // When wall to the side of player's origin square is not walkable (1st square from player square)
+  const coordinatesWallNearPlayer: Coordinates = {
+    y: startFromHorizontal ? coordinatesPlayer.y : coordinateOffsetSquare(coordinatesSquare.y, coordinatesPlayer.y, 1),
+    x: startFromHorizontal ? coordinateOffsetSquare(coordinatesSquare.x, coordinatesPlayer.x, 1) : coordinatesPlayer.x
+  };
+  const wallNearPlayer = getSquareByCoordinates(coordinatesWallNearPlayer, board);
+  if (wallNearPlayer.type !== SquareType.Wall) throw new Error("Square is not of type 'Wall'");
+  if (!wallNearPlayer.isWalkable) return false;
+
+  // When player square to the side of player's origin is not an enemy
+  const coordinatesEnemy: Coordinates = {
+    y: startFromHorizontal ? coordinatesPlayer.y : coordinatesSquare.y,
+    x: startFromHorizontal ? coordinatesSquare.x : coordinatesPlayer.x
+  };
+  const enemy = getSquareByCoordinates(coordinatesEnemy, board);
+  if (enemy.type !== SquareType.Player) throw new Error("Square is not of type 'Player'");
+  if (!enemy.playerId) return false;
+
+  // When wall between the targeted square and the enemy square is not walkable (adds 1 offset to coordinates in opposite cases than the wall near the player)
+  const coordinatesWallBetweenTargetAndEnemy: Coordinates = {
+    y: startFromHorizontal ? coordinateOffsetSquare(coordinatesSquare.y, coordinatesPlayer.y, 1) : coordinatesSquare.y,
+    x: startFromHorizontal ? coordinatesSquare.x : coordinateOffsetSquare(coordinatesSquare.x, coordinatesPlayer.x, 1)
+  };
+  const wallBetweenTargetAndEnemy = getSquareByCoordinates(coordinatesWallBetweenTargetAndEnemy, board);
+  if (wallBetweenTargetAndEnemy.type !== SquareType.Wall) throw new Error("Square is not of type 'Wall'");
+  if (!wallBetweenTargetAndEnemy.isWalkable) return false;
+
+  // When targeted square is not near the border of the board or there is no wall that would block linear jump (3rd square from player square)
+  const LAST_ROW_INDEX = BOARD_WIDTH - 1;
+
+  const coordinatesWallOverTheEnemy: Coordinates = {
+    y: startFromHorizontal ? coordinatesPlayer.y : coordinateOffsetSquare(coordinatesSquare.y, coordinatesPlayer.y, 3),
+    x: startFromHorizontal ? coordinateOffsetSquare(coordinatesSquare.x, coordinatesPlayer.x, 3) : coordinatesPlayer.x
+  };

```

Comment:
Create a variable called `isEnemyOnBoardEdge` and put that inside the `if` statement, so that it's clear what exactly you are checking here. Do `if (isEnemyOnBoardEdge) return true` and then continue the `wallOverTheEnemy` checks outside the `if` statement. This may look unnecessary, but when reading code, it's easier to understand when you can put code into words. In this case: "When enemy is on board edge, the player can hop on the target square`